subsys/fw_info/CMakeLists.txt

```diff
@@ -25,7 +25,8 @@ math(EXPR
    (${CONFIG_FW_INFO_MAGIC_COMPATIBILITY_ID} << 24)"
   )
 
-set(FIRMWARE_INFO_MAGIC   "${CONFIG_FW_INFO_MAGIC_COMMON},${CONFIG_FW_INFO_MAGIC_FIRMWARE_INFO},${MAGIC_COMPATIBILITY_INFO}")
+# This variable is used elsewhere, so make it globally available
+set(FIRMWARE_INFO_MAGIC   "${CONFIG_FW_INFO_MAGIC_COMMON},${CONFIG_FW_INFO_MAGIC_FIRMWARE_INFO},${MAGIC_COMPATIBILITY_INFO}" CACHE INTERNAL "")
```

Comment:
we should minimize polluting the CMake cache with data just because you need a global variable.
If a user starts with `CONFIG_FW_INFO=y` and then later reconfigures to `CONFIG_FW_INFO=n` then this value stays in the cache.

Could you change the library into a named library, like:
```
zephyr_library_named(ncs_fw_info)
```
and then add the `FIRMWARE_INFO_MAGIC` as a property there ?
So that the value is located together with it's use.
```
set_property(TARGET ncs_fw_info PROPERTY FIRMWARE_INFO_MAGIC <value>)
```